### PR TITLE
[Juniper SRX] Fix grok patterns for system logs

### PR DESCRIPTION
**Pull request by kcreddy**
## What does this PR do?
Fixes existing grok patterns for system logs

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [X] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [x] Waiting for the user who raised the request to get back on the issue comment. There are several instances where logs are not correctly formatted, or deviating from the standard.
Examples from `test-system.log`: 

    - line 2: `, Traffic-selector: FC Name:` has no comma between fields
    - line 3: `, VR-ID: 5: Role: Initiator` has `:` instead of `,` for field separator
    - line 6: `, VPN: IPSEC-NIKON-TUN1-PROD-VPN Gateway: IKE-NIKON-TUN1-GW, ` has ` ` instead of `,` for field separator. Also present in multiple places.
Confirmed by user that these can be ignored - https://github.com/elastic/integrations/issues/6963#issuecomment-1672306810

## Related issues

- Closes https://github.com/elastic/integrations/issues/6963

### CODE REVIEW

File: packages/juniper_srx/data_stream/log/elasticsearch/ingest_pipeline/system.yml [hunk 1]
```diff
@@ -8,102 +8,177 @@ processors:
...
-# Converts all kebab-case key names to snake_case
-- script:
-    lang: painless
-    if: ctx.juniper?.srx != null
-    source: >-
-        ctx.juniper.srx = ctx?.juniper?.srx.entrySet().stream().collect(Collectors.toMap(e -> e.getKey().replace('-', '_'), e -> e.getValue()));
 
+# Parse different types of system_structured_brief messages
+- grok:
+    if: "ctx._temp_?.unparsed?.system_structured_brief != null && ctx._temp_?.unparsed?.system_structured_brief != ''"
```
Comment (chemamartinez) [inline packages/juniper_srx/data_stream/log/elasticsearch/ingest_pipeline/system.yml:18]: Here we could avoid checking if the field exists for the second term
```suggestion
    if: "ctx._temp_?.unparsed?.system_structured_brief != null && ctx._temp_.unparsed.system_structured_brief != ''"
```

File: packages/juniper_srx/data_stream/log/elasticsearch/ingest_pipeline/system.yml [hunk 2]
```diff
@@ -8,102 +8,177 @@ processors:
...
 - set:
     field: juniper.srx.tag
-    value: "{{{tag}}}"
-    if: 'ctx.tag != null && (ctx.juniper?.srx?.tag == null || ctx.juniper?.srx?.tag == "-")'
-- set:
-    field: juniper.srx.tag
-    value: "{{{_temp_.sub_tag}}}"
-    if: 'ctx._temp_?.sub_tag != null && (ctx.juniper?.srx?.tag == null || ctx.juniper?.srx?.tag == "-")'
+    value: "{{{_temp_.tag_brief}}}"
+    if: 'ctx._temp_?.tag_brief != null && (ctx.juniper?.srx?.tag == null || ctx.juniper?.srx?.tag == "-")'
```
Comment (chemamartinez) [inline packages/juniper_srx/data_stream/log/elasticsearch/ingest_pipeline/system.yml:44]:
```suggestion
    if: 'ctx._temp_?.tag_brief != null && (ctx.juniper?.srx?.tag == null || ctx.juniper.srx.tag == "-")'
```

**Review by chemamartinez — APPROVED** [review]
I've added two small suggestions but overall it looks good to me.

File: packages/juniper_srx/data_stream/log/elasticsearch/ingest_pipeline/system.yml (hunk 2 above)
Comment (kcreddy) [reply packages/juniper_srx/data_stream/log/elasticsearch/ingest_pipeline/system.yml]: updated as suggested

File: packages/juniper_srx/data_stream/log/elasticsearch/ingest_pipeline/system.yml (hunk 1 above)
Comment (kcreddy) [reply packages/juniper_srx/data_stream/log/elasticsearch/ingest_pipeline/system.yml]: updated as suggested

**Comment by kcreddy** [comment]
Thanks @chemamartinez!
I think Marius might also want to take look. Lets wait for couple more days and merge once he is back and reviewed it.

File: packages/juniper_srx/data_stream/log/elasticsearch/ingest_pipeline/default.yml [hunk 3]
```diff
@@ -13,12 +13,15 @@ processors:
...
+      # 1. SRX Traffic structured log pattern
+      - '^<%{POSINT:syslog_pri}>(\d{1,3}\s)?(?:%{TIMESTAMP_ISO8601:_temp_.raw_date})\s%{SYSLOGHOST:syslog_hostname}\s%{JUNIPER_TRAFFIC_PROCESS:syslog_program}\s(?:%{POSINT:syslog_pid}|-)?\s%{WORD:tag}\s\[([^=]+?\s)?%{GREEDYDATA:_temp_.traffic_structured}\]\s?$'
+      # 2. SRX System structured log pattern (captures all structured logs when syslog_program not in JUNIPER_TRAFFIC_PROCESS)
+      - '^<%{POSINT:syslog_pri}>(\d{1,3}\s)?(?:%{CUSTOM_DATE:_temp_.raw_date})\s%{SYSLOGHOST:syslog_hostname}\s%{PROG:syslog_program}\s(?:%{POSINT:syslog_pid}|-)?\s%{WORD:tag}\s\[([^=]+?\s)?%{GREEDYDATA:_temp_.system_structured}\](?!=)\s?%{DATA:_temp_.unparsed.message}\s?$'
+      # 3. SRX System structured-brief and unstructured log patterns (further parsing done in system.yml)
       - '^<%{POSINT:syslog_pri}>(\d{1,3}\s)?(?:%{CUSTOM_DATE:_temp_.raw_date})\s%{SYSLOGHOST:syslog_hostname}\s%{GREEDYDATA:_temp_.unparsed.message}$'
       pattern_definitions:
         CUSTOM_DATE: "%{TIMESTAMP_ISO8601}|(%{MONTH}%{SPACE}+%{MONTHDAY}%{SPACE}+%{TIME})"
+        JUNIPER_TRAFFIC_PROCESS: "RT_FLOW|RT_UTM|RT_IDP|RT_IDS|RT_AAMW|RT_SECINTEL"
```
Comment (efd6) [inline packages/juniper_srx/data_stream/log/elasticsearch/ingest_pipeline/default.yml:24]: Minor refactoring to simplify, and reduce unnecessary captures:
```suggestion
      # 1. SRX Traffic structured log pattern
      - '^%{SYSLOG_PREFIX}?%{TIMESTAMP_ISO8601:_temp_.raw_date}\s%{SYSLOGHOST:syslog_hostname}\s%{JUNIPER_TRAFFIC_PROCESS:syslog_program}\s(?:%{POSINT:syslog_pid}|-)?\s%{WORD:tag}\s\[([^=]+?\s)?%{GREEDYDATA:_temp_.traffic_structured}\]\s?$'
      # 2. SRX System structured log pattern (captures all structured logs when syslog_program not in JUNIPER_TRAFFIC_PROCESS)
      - '^%{SYSLOG_PREFIX}?%{CUSTOM_DATE:_temp_.raw_date}\s%{SYSLOGHOST:syslog_hostname}\s%{PROG:syslog_program}\s(?:%{POSINT:syslog_pid}|-)?\s%{WORD:tag}\s\[([^=]+?\s)?%{GREEDYDATA:_temp_.system_structured}\](?!=)\s?%{DATA:_temp_.unparsed.message}\s?$'
      # 3. SRX System structured-brief and unstructured log patterns (further parsing done in system.yml)
      - '^%{SYSLOG_PREFIX}?%{CUSTOM_DATE:_temp_.raw_date}\s%{SYSLOGHOST:syslog_hostname}\s%{GREEDYDATA:_temp_.unparsed.message}$'
      pattern_definitions:
        SYSLOG_PREFIX: '<%{POSINT:syslog_pri}>(?:\d{1,3}\s)'
        CUSTOM_DATE: '%{TIMESTAMP_ISO8601}|%{MONTH}%{SPACE}+%{MONTHDAY}%{SPACE}+%{TIME}'
        JUNIPER_TRAFFIC_PROCESS: 'RT_FLOW|RT_UTM|RT_IDP|RT_IDS|RT_AAMW|RT_SECINTEL'
```

File: packages/juniper_srx/data_stream/log/elasticsearch/ingest_pipeline/default.yml [hunk 4]
```diff
@@ -29,6 +32,15 @@ processors:
       ignore_missing: true
       ignore_failure: false
       trim_value: "\""
+# split Juniper-SRX fields
+  - kv:
+      field: _temp_.system_structured
+      field_split: " (?=[a-z0-9\\_\\-]+=)"
```
Comment (efd6) [inline packages/juniper_srx/data_stream/log/elasticsearch/ingest_pipeline/default.yml:38]:
```suggestion
      field_split: ' (?=[a-z0-9\_\-]+=)'
```
(similar above)

File: packages/juniper_srx/data_stream/log/elasticsearch/ingest_pipeline/default.yml [hunk 5]
```diff
@@ -29,6 +32,15 @@ processors:
...
+      value_split: "="
+      prefix: "juniper.srx."
+      ignore_missing: true
+      ignore_failure: false
+      trim_value: "\""
```
Comment (efd6) [inline packages/juniper_srx/data_stream/log/elasticsearch/ingest_pipeline/default.yml:43]:
```suggestion
      trim_value: '"'
```

Comment (kcreddy) [reply packages/juniper_srx/data_stream/log/elasticsearch/ingest_pipeline/default.yml]: updated as suggested

File: packages/juniper_srx/data_stream/log/elasticsearch/ingest_pipeline/default.yml (hunk 4 above)
Comment (kcreddy) [reply packages/juniper_srx/data_stream/log/elasticsearch/ingest_pipeline/default.yml]: updated as suggested. Thanks I never noticed that 👍🏼

File: packages/juniper_srx/data_stream/log/elasticsearch/ingest_pipeline/default.yml (hunk 3 above)
Comment (kcreddy) [reply packages/juniper_srx/data_stream/log/elasticsearch/ingest_pipeline/default.yml]: updated as suggested

**Review by efd6 — APPROVED** [review]

**Review by P1llus — APPROVED** [review]

**Pull request merged**